pkg/api/apis/operators/v1alpha1/clusterserviceversion_types.go

```diff
@@ -182,20 +182,23 @@ const (
 type ConditionReason string
 
 const (
-	CSVReasonRequirementsUnknown     ConditionReason = "RequirementsUnknown"
-	CSVReasonRequirementsNotMet      ConditionReason = "RequirementsNotMet"
-	CSVReasonRequirementsMet         ConditionReason = "AllRequirementsMet"
-	CSVReasonOwnerConflict           ConditionReason = "OwnerConflict"
-	CSVReasonComponentFailed         ConditionReason = "InstallComponentFailed"
-	CSVReasonInvalidStrategy         ConditionReason = "InvalidInstallStrategy"
-	CSVReasonWaiting                 ConditionReason = "InstallWaiting"
-	CSVReasonInstallSuccessful       ConditionReason = "InstallSucceeded"
-	CSVReasonInstallCheckFailed      ConditionReason = "InstallCheckFailed"
-	CSVReasonComponentUnhealthy      ConditionReason = "ComponentUnhealthy"
-	CSVReasonBeingReplaced           ConditionReason = "BeingReplaced"
-	CSVReasonReplaced                ConditionReason = "Replaced"
-	CSVReasonNeedCertRotation        ConditionReason = "NeedCertRotation"
-	CSVReasonAPIServiceResourceIssue ConditionReason = "APIServiceResourceIssue"
+	CSVReasonRequirementsUnknown              ConditionReason = "RequirementsUnknown"
+	CSVReasonRequirementsNotMet               ConditionReason = "RequirementsNotMet"
+	CSVReasonRequirementsMet                  ConditionReason = "AllRequirementsMet"
+	CSVReasonOwnerConflict                    ConditionReason = "OwnerConflict"
+	CSVReasonComponentFailed                  ConditionReason = "InstallComponentFailed"
+	CSVReasonInvalidStrategy                  ConditionReason = "InvalidInstallStrategy"
+	CSVReasonWaiting                          ConditionReason = "InstallWaiting"
+	CSVReasonInstallSuccessful                ConditionReason = "InstallSucceeded"
+	CSVReasonInstallCheckFailed               ConditionReason = "InstallCheckFailed"
+	CSVReasonComponentUnhealthy               ConditionReason = "ComponentUnhealthy"
+	CSVReasonBeingReplaced                    ConditionReason = "BeingReplaced"
+	CSVReasonReplaced                         ConditionReason = "Replaced"
+	CSVReasonNeedsReinstall                   ConditionReason = "NeedsReinstall"
+	CSVReasonNeedsCertRotation                ConditionReason = "NeedsCertRotation"
+	CSVReasonAPIServiceResourceIssue          ConditionReason = "APIServiceResourceIssue"
+	CSVReasonAPIServiceResourcesNeedReinstall ConditionReason = "APIServiceResourcesNeedReinstall"
```

Comment:
Are the [lifecycle docs](https://github.com/operator-framework/operator-lifecycle-manager/blob/master/Documentation/design/architecture.md#clusterserviceversion-control-loop) still correct once we add this change?

Reply:
I've updated them.